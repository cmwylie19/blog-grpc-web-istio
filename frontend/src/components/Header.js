import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import Logo from "../assets/logo-w.svg";

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
    marginBottom: "80px",
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
}));

export default function Header({ changeTheme }) {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="fixed">
        <Toolbar>
          <IconButton
            edge="start"
            className={classes.menuButton}
            color="inherit"
            aria-label="menu"
          >
            <img src={Logo} width="90" height="72" alt="solo.io" />
          </IconButton>
          <Typography
            variant="h6"
            className={classes.title}
            onClick={() => changeTheme()}
          ></Typography>
          <div>
            <Typography variant="overline" onClick={() => changeTheme()}>
              Change Theme
            </Typography>
          </div>
        </Toolbar>
      </AppBar>
    </div>
  );
}
