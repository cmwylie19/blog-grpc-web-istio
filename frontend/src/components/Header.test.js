import React from "react";
import { mount, shallow } from "enzyme";
import Header from "./Header";

let renderedComponent;
let mockSetAuth = () => true;

beforeEach(() => {
  renderedComponent = shallow(<Header first="test" />);
});

afterEach(() => {
  jest.resetAllMocks();
});

describe("SignUp Component", () => {
  it("should render compoennt", () => {
    expect(renderedComponent).toBeDefined();
  });

  it("Should match snapshot", () => {
    expect(renderedComponent).toMatchSnapshot();
  });
});
