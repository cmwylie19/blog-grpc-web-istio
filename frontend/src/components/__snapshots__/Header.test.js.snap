// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Component Should match snapshot 1`] = `ShallowWrapper {}`;
